src/main/java/school/lemon/changerequest/java/collections/hw1/ExtendedListImpl.java

```diff
@@ -0,0 +1,709 @@
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    @Override
+    public boolean contains(Object o) {
+        return indexOf(o) >= 0;
+    }
+
+    @Override
+    public Iterator<E> iterator() {
+        return new Itr(this);
+    }
+
+    @Override
+    public Object[] toArray() {
+        return Arrays.copyOf(container, size);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T> T[] toArray(T[] a) {
+        notNull(a);
+        if (a.length < size)
+            return (T[]) Arrays.copyOf(container, size, a.getClass());
+        System.arraycopy((T[]) container, 0, a, 0, size);
+        if (a.length > size)
+            a[size] = null;
+        return a;
+    }
+
+    @Override
+    public boolean add(E e) {
+        add(size, e);
+        return true;
+    }
+
+    @Override
+    public boolean remove(Object o) {
+        int index = indexOf(o);
+        if (index >= 0) {
+            remove(index);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean containsAll(Collection<?> c) {
+        notNull(c);
+        for (Object o : c)
+            if (!contains(o))
+                return false;
+        return true;
+    }
+
+    @Override
+    public boolean addAll(Collection<? extends E> c) {
+        notNull(c);
+        return addAll(size, c);
+    }
+
+    @Override
+    public boolean addAll(int index, Collection<? extends E> c) {
+        notNull(c);
+        isCorrectIndexForAdd(index);
+        int cSize = c.size();
+        if (cSize == 0)
+            return false;
+        if (size + cSize > container.length) {
+            int newSize = container.length;
+            do {
+                newSize *= 2;
+            }
+            while (newSize < size + cSize);
+            Object[] tmp = new Object[newSize];
+            System.arraycopy(container, 0, tmp, 0, index);
+            System.arraycopy(c.toArray(), 0, tmp, index, cSize);
+            System.arraycopy(container, index, tmp, index + cSize, size - index);
+            container = tmp;
+        } else {
+            System.arraycopy(container, index, container, index + cSize, size - index);
+            System.arraycopy(c.toArray(), 0, container, index, cSize);
+        }
+        size += cSize;
+        modificationCoef++;
+        return true;
+    }
+
+    @Override
+    public boolean removeAll(Collection<?> c) {
+        notNull(c);
+        boolean result = false;
+        for (int i = 0; i < size; i++)
+            if (c.contains(container[i])) {
+                remove(i);
+                i--;
+                result = true;
+            }
+        if (result)
+            modificationCoef++;
+        return result;
+    }
+
+    @Override
+    public boolean retainAll(Collection<?> c) {
+        notNull(c);
+        boolean result = false;
```

Comment:
This name doesn't show any real context here. I believe something like `isChanged` or `isModified` could be better here.